### PR TITLE
Use socketio server instead of uwsgi

### PR DESCRIPTION
**Pull request by clemensmanert**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,27 +113,56 @@ pip install -r "requirements.txt"
 pip install git+https://github.com/buzzn/discovergy.git
 ```
 
-Create an endpoint in your webserver, for example for Apache, create a new site:
+Now create a systemd service `mybuzzn-backend.service`, which starts the socketio server we use to run
+the app in:
+```ini
+[Unit]
+Description=Run mybuzzn-backend as an flask socketio instance.
```
Comment (missgreenwood) [inline README.md:120]: "Description=Run mybuzzn-backend as a flask socketio instance."

File: README.md [hunk 2]
```diff
@@ -113,27 +113,56 @@ pip install -r "requirements.txt"
...
+Group=mybuzznbackend
+
+[Install]
+WantedBy=multi-user.target
 ```
-<VirtualHost *:80>
-  ServerName your-domain.net
-  ErrorLog /path/to/error/log
-  CustomLog /path/to/access.log combined
+Either store it on `/etc/systemd/system/` or create there a link pointing to it.
```
Comment (missgreenwood) [inline README.md]: "Either store it on `/etc/systemd/system/` or create a link there pointing to it."

File: README.md [hunk 3]
```diff
@@ -180,10 +210,10 @@ WantedBy=multi-user.target⏎
 ### Upgrading to a new version
 To upgrade to a new version, go to the project root and run `git pull`.
 If something has changed on the models, run `source ./venv/bin/activate` to
```
Comment (missgreenwood) [inline README.md]: Maybe clarify you are talking about the database models: 
"If something has changed in the database models..."

File: README.md [hunk 4]
```diff
@@ -180,10 +210,10 @@ WantedBy=multi-user.target⏎
...
-activate the python environment. Then run `./venv/bin/flask db upgrade` to
+activate the python environment. Then run `./flask db upgrade` to
 upgrade the database.
 
 Finally restart the redis task, which fills the redis database with meter
 readings and the webserver:
```
Comment (missgreenwood) [inline README.md]: Maybe a little clearer: "Finally restart the the web server and the redis task which fills the redis database with meter
readings:"

File: flask [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/python3
+
+import re
+import sys
+import setup_environment
+
+from flask.cli import main
+if __name__ == '__main__':
+    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
+    sys.exit(main())
```
Comment (missgreenwood) [inline flask]: What is the purpose of this script?

File: mybuzzn-backend.py [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import setup_environment
+import app
+
+app.run_server()
```
Comment (missgreenwood) [inline mybuzzn-backend.py:6]: Likewise, what is the purpose of this script?

File: mybuzzn-backend.service [hunk 7]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=Run mybuzzn-backend as an flask socketio instance.
```
Comment (missgreenwood) [inline mybuzzn-backend.service]: Like above: "Description=Run mybuzzn-backend as a flask socketio instance."

File: start_redis_task.py [hunk 8]
```diff
@@ -1,9 +1,5 @@
-activate_this = '/var/www/mybuzzn-backend.buzzn.net/mybuzznbackend/env/bin/activate_this.py'
-with open(activate_this) as file_:
-    exec(file_.read(), dict(__file__=activate_this))
-
 import setup_environment
-    
 import util.task
+
 util.task.run()
```
Comment (missgreenwood) [inline start_redis_task.py:4]: Could you explain why you made these changes, please?

**Review by missgreenwood — APPROVED** [review]

File: mybuzzn-backend.py (hunk 6 above)
Comment (clemensmanert) [reply mybuzzn-backend.py:6]: I added a comment.

File: start_redis_task.py (hunk 8 above)
Comment (clemensmanert) [reply start_redis_task.py:4]: The setup is done in the setup_envornment. The service will execute this script in the virtualenv, so the script does not create the virtualenv anymore.

File: flask (hunk 5 above)
Comment (clemensmanert) [reply flask]: I added a comment in the script.

File: mybuzzn-backend.py (hunk 6 above)
Comment (clemensmanert) [reply mybuzzn-backend.py:6]: I added a comment in the script.

File: start_redis_task.py (hunk 8 above)
Comment (clemensmanert) [reply start_redis_task.py:4]: The service will start the script in a virtual env, so there is no need for the script to setup the virtualenv.

**Pull request merged**